doc: README.rst added to subsys/console/echo sample (#64626)
zephyrproject-rtos/zephyr: samples/subsys/console/echo/README.rst

.. zephyr:code-sample:: console_echo

>    :name: console echo

Reviewer: ```suggestion
   :name: Console echo
```

   :relevant-api: console_api

   Echo an input character back to the output using the Console API.

Overview
********

This example shows how the :c:func:`console_getchar` function and

> :c:func:`console_putchar` function can be used to echo an input character

Reviewer: ```suggestion
:c:func:`console_putchar` functions can be used to echo an input character
```

back to the console.

Requirements
************

> UART console is required to run this sample.

Author: @kartben I saw you mentioned this should change - I am perhaps referring mistakenly to the terminal program I am using here (picocom). This is also consistent with the other console samples.

Reviewer: Ya I removed my comment as I was mostly getting confused between shell and console :) I don't think a console that's *not* UART makes sense so nevermind :)

> UART console is required to run this sample.

Reviewer: Similar to an earlier comment regarding being unnecessarily uart-specific, isn't the requirement rather: "A board that defines a `zephyr,console`."?

Building and Running
********************

Build and flash the sample as follows, changing ``nucleo_f401re`` for your
board. Alternatively you can run this using QEMU, as described in
:zephyr:code-sample:`console_getchar`.

.. zephyr-app-commands::
   :zephyr-app: samples/subsys/console/echo
   :host-os: unix
   :board: nucleo_f401re
   :goals: build flash
   :compact:

> Following the initial prompt given by the firmware start pressing keys on a

Reviewer: ```suggestion
Following the initial prompt given by the firmware, start pressing keys on a
```

.. zephyr:code-sample:: console_echo
   :name: console echo

>    :relevant-api: console_api

Reviewer: You would still need to add a new page under https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/services/index.rst
It's OK if for now it only contains the API Reference entry, similar to https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/services/poweroff.rst

>    :relevant-api: console_api

Author: Oh right, ok - I'll add a page. Thanks Ben!

>    :relevant-api: console_api

Author: Should be ok now. I'm not sure to inspect the build to ensure it has rendered correctly however.

>    :relevant-api: console_api

Reviewer: > Should be ok now. I'm not sure to inspect the build to ensure it has rendered correctly however.

Looking good, yes :) 

https://builds.zephyrproject.io/zephyr/pr/64626/docs/services/console.html
https://builds.zephyrproject.io/zephyr/pr/64626/docs/samples/subsys/console/echo/README.html#console_echo

>    Use console_getchar() and console_putchar() to echo an input character back
>    to the output.

Reviewer: Focus on end result:

```suggestion
   Echo an input character back to the output using the Console API.
```

Overview
********

This example shows how the :c:func:`console_getchar` function and
:c:func:`console_putchar` function can be used to echo an input character

> to a UART console.

Author: Good point, I'll make it more generic - so I suppose `usart2` could be something else entirely?

`	model = "STMicroelectronics STM32F401RE-NUCLEO board";
	compatible = "st,stm32f401re-nucleo";

	chosen {
		zephyr,console = &usart2;
		zephyr,shell-uart = &usart2;
		zephyr,sram = &sram0;
		zephyr,flash = &flash0;
		zephyr,code-partition = &slot0_partition;
	};`
